### PR TITLE
refact: Use Data Interface, refactoring i18n module.

### PR DESCRIPTION
**Pull request by 2jun0**
# Use a Data interface 0be70f2134d1ad96eb4f347f050432131731d764
add Data interface. it'll be used to stored in storage or exchanged in popup, content and background

# Refactoring i18n fadf51fdcbebc33226ffca72c8ed9aba76463251
- The browser locale value was handled at `i18n`
- The `localize` function has a `locale` variable
- The localeJson's keys value(locales) was handled at `i18n`

# Fix small thing. c28752c48a40e42faea6caff7a8526d133711efe
- changed `tabs` to `tabs.length` (tabs is always `[]`)

### CODE REVIEW

**Comment by Heroor** [comment]
It is recommended that you rearrange the code and submit the PR, removing the meaningless commit.

**Comment by Heroor** [comment]
Rebase can be used when merging code for the main repository to reduce the merge type commit.

File: src/core/data.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import MD_PLUGINS from '../config/md-plugins'
+import PAGE_THEMES from '../config/page-themes'
+import i18n from '../popup/i18n'
+
+interface Data {
+  enable?: boolean
+  refresh?: boolean
+  pageTheme?: typeof PAGE_THEMES[0]
+  locale?: string
```
Comment (Heroor) [inline src/core/data.ts:9]: After a field is changed, the field stored by an installed user becomes invalid.

File: src/core/data.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import MD_PLUGINS from '../config/md-plugins'
+import PAGE_THEMES from '../config/page-themes'
+import i18n from '../popup/i18n'
+
+interface Data {
+  enable?: boolean
+  refresh?: boolean
+  pageTheme?: typeof PAGE_THEMES[0]
```
Comment (Heroor) [inline src/core/data.ts:8]: After a field is changed, the field stored by an installed user becomes invalid.

File: src/core/data.ts (hunk 1 above)
Comment (Heroor) [inline src/core/data.ts:9]: After a field is changed, the field stored by an installed user becomes invalid.

File: src/core/data.ts [hunk 3]
```diff
@@ -6,17 +6,17 @@ interface Data {
   enable?: boolean
   refresh?: boolean
   pageTheme?: typeof PAGE_THEMES[0]
-  locale?: string
-  mdPlugins?: typeof MD_PLUGINS
+  language?: string
+  selectedMdPlugins?: typeof MD_PLUGINS
```
Comment (2jun0) [inline src/core/data.ts]: I'm sorry 😥
I didn't think about the users who is using it now.

I had data field reverted to previous

**Review by Heroor — COMMENTED** [review]
fix some bug

**Pull request merged**